<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>

  <!-- Config and weights folder. -->
  <arg name="yolo_weights_path"          default="$(find cubicle_detect)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find cubicle_detect)/yolo_network_config/cfg"/>
  <arg name="image_topic"                default="/zed/left/image_rect_color"/>
  <arg name="image_info"                 default="/zed/left/camera_info"/>
  <arg name="disparity_topic"            default="/zed/disparity/disparity_image"/>
  <arg name="disparity_scope"            default="128"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="cubicle_detect" file="$(find cubicle_detect)/config/zed.yaml"/>
  <rosparam command="load" ns="cubicle_detect" file="$(find cubicle_detect)/config/yolov3_bdd.yaml"/>

  <!-- Start darknet and ros wrapper -->
  <node pkg="cubicle_detect" type="cubicle_detect" name="cubicle_detect" output="screen" launch-prefix="$(arg launch_prefix)">
    <!--whether use one-channel as input-->
    <param name="use_grey"              value="false" />
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <param name="disparity_scope"       value="$(arg disparity_scope)" />
    <param name="image_topic"           value="$(arg image_topic)" />
    <param name="image_info"            value="$(arg image_info)" />
    <param name="disparity_topic"       value="$(arg disparity_topic)" />
    <param name="scale"                 value="1" />
  </node>

  <!--<include file="$(find zed_wrapper)/launch/zed.launch">-->
  <!--</include>-->

 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->
</launch>
